Modifies AutomotiveSimulator to Allow Client Selectable Model Files.

liangfok opened the pull request:
Builds upon #3933. Goes towards spiral 1 of #3919.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/3959)
<!-- Reviewable:end -->

liangfok commented:
+@maddog-tri for feature review please.
+@jwnimmer-tri for platform review please.

Thanks!

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVXJIKekgUstm9h9tCJ:b-kigsmr)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KVXTMAg3UWQRZ6ithb0:b-d9jm7n) (no related file):*
(N.B. New discussion; don't mix the threads.)

I am disappointed that the automotive demo is being infected with even more SDF model/joint hoopla.  That we can't simply visualize a different mesh while using simple car kinematics without jumping through joint arithmetic loops is an incredible defect in our software architecture.  That we can't _unit test_ the automotive system simulator without also reading in SDF goop and testing based on derived joint arithmetic is even more terrible.  Supposedly the "geometry world" silver bullet will slay all of these monsters, but I remain skeptical.  I hope I am proven wrong.

I guess there's not much to do in response to this discussion, but if there is any way to make this ever-so-simple concept make sense with less code, I'd be all ears.  I wonder, what if we created an SDF that was just the Prius mesh(es), without any joints?  Wouldn't it work just as simply as boxcar.urdf did, yet give us the code simplicity that I'm desiring?

Maybe the answer is to just suspend #3919 work after this PR, and iterate on a better "geometry world" solution with high priority instead.

---

*[drake/automotive/automotive_simulator.cc, line 115 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXSKFPiwS2HuEV8zw4:-KVXSKFPiwS2HuEV8zw5:b-3w1v1k) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.cc#L115)):*
> ```cpp
>   const int model_instance_id = table.begin()->second;
>   const std::vector<const RigidBody*> bodies =
>       rigid_body_tree_->FindModelInstanceBodies(model_instance_id);
> ```

I think `bodies` is unused.

---

*[drake/automotive/automotive_simulator.h, line 54 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXRCLSCv4MRBMc8ruR:-KVXRCLSCv4MRBMc8ruS:b-mkubz8) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L54)):*
> ```objc
>   /// @pre Start() has NOT been called.
>   ///
>   /// @param[in] sdf_file The name of the SDF file to load for the simple car.
> ```

BTW Clarify that the model file is only for visualization, and not for physical simulation?

---

*[drake/automotive/automotive_simulator.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXQrpHcsNdRBcpLhIz:-KVXQrpHcsNdRBcpLhJ-:budvsfk) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L73)):*
> ```objc
>   ///
>   /// @param[in] start_time The time at which the trajectory car should start
>   /// moving.
> ```

BTW For the above three parameters, consider merely referring the reader to the TrajectoryCar constructor documentation, instead of brittlely duplicating its content here.

---

*[drake/automotive/automotive_simulator.h, line 133 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXS3LDCYeuZOHVWMPK:-KVXS3LDCYeuZOHVWMPL:b-5cjznb) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L133)):*
> ```objc
>  private:
>   int allocate_vehicle_number();
>   int AddSdfModel(const std::string& sdf_filename,
> ```

Please be consistent with `sdf_file` vs `sdf_filename` within this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVXVja3RrxlJqDn-vcQ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KVX_do5pxpxBg-R-Quj:b74r35c) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> (N.B. New discussion; don't mix the threads.)
> 
> I am disappointed that the automotive demo is being infected with even more SDF model/joint hoopla.  That we can't simply visualize a different mesh while using simple car kinematics without jumping through joint arithmetic loops is an incredible defect in our software architecture.  That we can't _unit test_ the automotive system simulator without also reading in SDF goop and testing based on derived joint arithmetic is even more terrible.  Supposedly the "geometry world" silver bullet will slay all of these monsters, but I remain skeptical.  I hope I am proven wrong.
> 
> I guess there's not much to do in response to this discussion, but if there is any way to make this ever-so-simple concept make sense with less code, I'd be all ears.  I wonder, what if we created an SDF that was just the Prius mesh(es), without any joints?  Wouldn't it work just as simply as boxcar.urdf did, yet give us the code simplicity that I'm desiring?
> 
> Maybe the answer is to just suspend #3919 work after this PR, and iterate on a better "geometry world" solution with high priority instead.
</details>
Yes, I agree. I think the fundamental problem is the level of abstraction provided by `RigidBodyTree` (e.g., rigid bodies and joints) is too low for our needs. Presumably a "GeometryWorld" or "SharedEnvironment" will wrap a `RigidBodyTree` and other low-level abstractions like sensors / actuators and present object-level abstractions (e.g., "a car", "a pedstrian", "a building", etc.) . I'd be happy to suspend #3919 and instead focus on working with @seancurtis-tri to speed up the development of this higher-level abstraction.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
@jwnimmer-tri I addressed your comments.

---

Review status: 1 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/automotive_simulator.cc, line 115 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXSKFPiwS2HuEV8zw4:-KVXj3HbTGd2rhu1ry7W:boyexbg) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.cc#L115)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> I think `bodies` is unused.
</details>
Done. Thanks.

---

*[drake/automotive/automotive_simulator.h, line 54 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXRCLSCv4MRBMc8ruR:-KVXjo9yKwxmQEscdWYa:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L54)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> BTW Clarify that the model file is only for visualization, and not for physical simulation?
</details>
Done.

---

*[drake/automotive/automotive_simulator.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXQrpHcsNdRBcpLhIz:-KVXk50r_Ssck15KyTuC:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L73)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> BTW For the above three parameters, consider merely referring the reader to the TrajectoryCar constructor documentation, instead of brittlely duplicating its content here.
</details>
Done.

---

*[drake/automotive/automotive_simulator.h, line 133 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXS3LDCYeuZOHVWMPK:-KVXkeXVadL5TucoKvZq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c8240780194c56347085703e6b63da10393b7c4/drake/automotive/automotive_simulator.h#L133)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> Please be consistent with `sdf_file` vs `sdf_filename` within this file.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVXkgJaCWJ3Pc0PRzAx:b-ev77i5)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_Kdi9o8hIqo-XKCyp:b-afwtjr) (no related file):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Yes, I agree. I think the fundamental problem is the level of abstraction provided by `RigidBodyTree` (e.g., rigid bodies and joints) is too low for our needs. Presumably a "GeometryWorld" or "SharedEnvironment" will wrap a `RigidBodyTree` and other low-level abstractions like sensors / actuators and present object-level abstractions (e.g., "a car", "a pedstrian", "a building", etc.) . I'd be happy to suspend #3919 and instead focus on working with @seancurtis-tri to speed up the development of this higher-level abstraction.
</details>
BTW My first thought on this is that you may be disappointed.  As of my current knowledge, geometry world/shared environment is about one thing: allowing disparate systems to interact at the geometric level.  For example, it allows a sensor to see the geometry of a dummy car driving by...whatever, the geometry driven by a rigid body plant, and the static world.  This contrasts with the current state, for example, where the rigid body tree owns its own geometry and is the only entity able to make geometric queries on that geometry.

However, the geometry world is only a common, unifying interface.  It is agnostic of how the frames for the geometry get updated w.r.t. time.  So, if you have a demo which needs to load geometry and drive it using a rigid body tree, Geometry world won't lessen that at all.  If @sherm1 has a different view on this, he should chime in as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_OxB2_GKAIgvIR4Ft:b-tqd3ij) (no related file):*
@seancurtis-tri: Interesting. It sounds like `GeometryWorld`will not provide the object-level abstractions that we need. Instead, it sounds like a class that unifies multiple sources of geometry information (like combining multiple `RigidBodyTree` objects each representing different things). 

In this case, maybe we need a `World` abstraction that encapsulates both `GeometryWorld` and `RigidBodyTree` and provides the desired object-level abstractions. It'll basically be responsible for doing the bookkeeping necessary to identify different branches of a `RigidBodyTree` (or multiple `RigidBodyTree` objects) as belonging to distinct objects like vehicles, pedestrians, etc., alleviating the need for the client to perform tedious rigid body and joint index math.

Quote from @jwnimmer-tri:

> I wonder, what if we created an SDF that was just the Prius mesh(es), without any joints? Wouldn't it work just as simply as boxcar.urdf did, yet give us the code simplicity that I'm desiring?

Doing this will result in the chassis showing up without wheels because the wheels are not part of the chassis mesh. Obviously, for #3919 spiral 2, this will also prevent the wheels (if they were to appear) from rotation and steering.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_Sf4pFaDrkINUagSe:b-iy6kae) (no related file):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> @seancurtis-tri: Interesting. It sounds like `GeometryWorld`will not provide the object-level abstractions that we need. Instead, it sounds like a class that unifies multiple sources of geometry information (like combining multiple `RigidBodyTree` objects each representing different things). 
> 
> In this case, maybe we need a `World` abstraction that encapsulates both `GeometryWorld` and `RigidBodyTree` and provides the desired object-level abstractions. It'll basically be responsible for doing the bookkeeping necessary to identify different branches of a `RigidBodyTree` (or multiple `RigidBodyTree` objects) as belonging to distinct objects like vehicles, pedestrians, etc., alleviating the need for the client to perform tedious rigid body and joint index math.
> 
> Quote from @jwnimmer-tri:
> 
> > I wonder, what if we created an SDF that was just the Prius mesh(es), without any joints? Wouldn't it work just as simply as boxcar.urdf did, yet give us the code simplicity that I'm desiring?
> 
> Doing this will result in the chassis showing up without wheels because the wheels are not part of the chassis mesh. Obviously, for #3919 spiral 2, this will also prevent the wheels (if they were to appear) from rotation and steering.
</details>
@liangfok Yep, you've basically got the idea.  The SharedEnvironment concept extends that further to include environmental factors (e.g., fog) to affect relevant queries (e.g., ray casting).  Essentially, it will register geometry (and geometry properties) to frames.  The time-dependent state of the frame is driven from an input port.  So, whether the frame is updated as part of a rigid body tree, part of a kinematics method, or based on random noise, GeometryWorld doesn't know.  It'll just position the frame and make queries on the geometric state of the world at any given time.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_TebFcWl3MvkGlJiA:b-ezdkfm) (no related file):*
> Doing this will result in the chassis showing up without wheels because the wheels are not part of the chassis mesh. Obviously, for #3919 spiral 2, this will also prevent the wheels (if they were to appear) from rotation and steering.

@liangfok Why can't the joints (and mutiple meshes) remain, just change to fixed (moving the DOFs, and thus the extra zeros all over the place)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_Tqexd3CwTUmIsIsU:bavgkn0) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> > Doing this will result in the chassis showing up without wheels because the wheels are not part of the chassis mesh. Obviously, for #3919 spiral 2, this will also prevent the wheels (if they were to appear) from rotation and steering.
> 
> @liangfok Why can't the joints (and mutiple meshes) remain, just change to fixed (moving the DOFs, and thus the extra zeros all over the place)?
</details>
And on the bigger picture of doing this all better, David and I chatted offline and we'll work on a write-up later in the day.  The upthread discussions don't really capture my core complaints.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_UbqMVPmEpSQnR_L8:bhfklb3) (no related file):*
> > > I wonder, what if we created an SDF that was just the Prius mesh(es), without any joints? Wouldn't it work just as simply as boxcar.urdf did, yet give us the code simplicity that I'm desiring?
> >
> > Doing this will result in the chassis showing up without wheels because the wheels are not part of the chassis mesh. Obviously, for #3919 spiral 2, this will also prevent the wheels (if they were to appear) from rotation and steering.
> 
> 
> @liangfok Why can't the joints (and mutiple meshes) remain, just change to fixed (moving the DOFs, and thus the extra zeros all over the place)?

@jwnimmer-tri, in your original statement, I mistook "without any joints" to mean "not even fixed joints". Yes, changing the joints to be fixed will eliminate the DOFs and thus eliminate the need to specify their states. This will only work in #3919 spiral 1. It will obviously not allow accomplishing #3919 spiral 2 since that requires that the fixed joints holding the wheels and steering arms be non-fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_VY_NO8sHs2pPEc1E:b-x0n5vn) (no related file):*
> This will only work in #3919 spiral 1. It will obviously not allow accomplishing #3919 spiral 2 since that requires that the fixed joints holding the wheels and steering arms be non-fixed.

Right.  And if the plan for Spiral 2 is to use an _entirely different_ technology base, keeping Spiral 1 simpler would have been to our advantage.  At this point though the work is done, and I think it's fine to merge this as usual.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVXTMAe2ATstipT9XW5:-KV_WImZfNWS6UAzONG_:b-wrx0bw) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> > This will only work in #3919 spiral 1. It will obviously not allow accomplishing #3919 spiral 2 since that requires that the fixed joints holding the wheels and steering arms be non-fixed.
> 
> Right.  And if the plan for Spiral 2 is to use an _entirely different_ technology base, keeping Spiral 1 simpler would have been to our advantage.  At this point though the work is done, and I think it's fine to merge this as usual.
</details>
Alright. I look forward to your thoughts on the right technologies / APIs for achieving #3919 spiral 2.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[drake/automotive/automotive_demo.cc, line 29 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeoFXo8ieqnVPsX6ne:-KVeoFXpbWESGtVXt_u0:b2q0tzt) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_demo.cc#L29)):*
> ```cpp
>   }
> 
>   const std::string sdf_file =
> ```

Consider using the `kSdfFile` naming convention for this constant (not strictly required by style-guide since it's not static storage). The filename cannot be changed (e.g., supplied by command-line arg) without considering other hard-coded assumptions about the model's geometry (in particular, size and float-joint location --- e.g., CreateTrajectoryParams() expects a "car" to have a particular length/width).  Hmm... no matter the name of this variable, it should have a warning comment about changing the value.

---

*[drake/automotive/automotive_simulator.h, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3Eo2RaPbPttCHu3w:-KVf3Eo2RaPbPttCHu3x:bf8g5il) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L55)):*
> ```objc
>   ///
>   /// @param[in] sdf_filename The name of the SDF file to load as the
>   /// visualization for the simple car.
> ```

There are conditions on the content of the SDF file that should be documented here, e.g., single model, only the first/floating joint is controlled (others set to zeros), etc.

---

*[drake/automotive/automotive_simulator.h, line 67 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3eQm_sCryobalS_6:-KVf3eQm_sCryobalS_7:b-n5mxg6) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L67)):*
> ```objc
>   ///
>   /// @param[in] sdf_filename The name of the SDF file to load as the
>   /// visualization for the trajectory car.
> ```

Same comment as on sdf_filename param in AddSimpleCar().

---

*[drake/automotive/automotive_simulator.h, line 77 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeyzPsgOUflhG6iMOT:-KVeyzPt8L0dbn5Msd7I:bgpw6ho) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L77)):*
> ```objc
>   /// @return The model instance ID of the TrajectoryCar that was just added to
>   /// the simulation.
>   int AddTrajectoryCarFromSdf(const std::string& sdf_filename,
> ```

Does the return value serve any purpose beyond the unit-tests?

---

*[drake/automotive/test/automotive_simulator_test.cc, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeww4yJLqiqyDacC-B:-KVeww4zntD1u6eZHtxi:b-ulwuky) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L37)):*
> ```cpp
> // name of the file containing the model of the vehicle to be used by the
> // SimpleCar.
> void TestSimpleCarWithSdf(const std::string& sdf_filename) {
> ```

This parameterization is a lie:  the test code still depends on the Prius SDF ...or, at least an SDF that has a "chassis floor".  Also, the parameterization and restructuring as a helper function is a detriment unless the test is meaningfully invoked on more than the single Prius SDF.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 43 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVetuabyT2Nyht9sUQI:-KVetuaclMo6aRb6c_rz:b7f5arz) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L43)):*
> ```cpp
>   const std::string driving_command_name =
>       systems::lcm::LcmSubscriberSystem::get_name(kCommandChannelName);
>   // TODO(jwnimmer-tri) Do something better than "0_" here.
> ```

This comment looks like it belongs with the definition of kJointStateChannelName.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 57 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeuDqnxlk1ELOrOJV1:-KVeuDqow2FAdjUEFWII:bd3d87u) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L57)):*
> ```cpp
>                                      .size();
> 
>   // Grab the pieces we want (testing the GetSystemByName in the process).
> ```

Stale comment?  E.g., s/GetSystemByName/GetBuilderSystemByName/  ?

---

*[drake/automotive/test/automotive_simulator_test.cc, line 70 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVexbA4mKtYyCznNRbX:-KVexbA4mKtYyCznNRbY:b-5bj08) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L70)):*
> ```cpp
>   EXPECT_EQ(1, tree.get_num_model_instances());
>   // One body belongs to the world, the rest belong to the car.
>   ASSERT_EQ(1 + num_vehicle_bodies, tree.get_num_bodies());
> ```

It's not completely clear to me what this is testing. Since `num_vehicle_bodies` came from a query to the tree, is it that we are making sure that AddSimpleCarFromSdf() didn't add something else?  It was more clear with `kNumVehicleBodies`; then the tester had an explicit expectation of what as in the SDF and what would end up in the RBT.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 134 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf4p7x8XOO1MzXy3ym:-KVf4p7x8XOO1MzXy3yn:b-gtqi51) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L134)):*
> ```cpp
> // the name of the file containing the model of the vehicle to be used by the
> // TrajectoryCar.
> void TestTrajectoryCarWithSdf(const std::string& sdf_file) {
> ```

Same comment as TestSimpleCarWithSdf():  making it a helper function is just more noise until more than a single SDF is actually meaningfully tested.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 158 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf-8Bt6tIdBUm775E2:-KVf-8Bu6b-9PTHYlNjs:b8p1l2u) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L158)):*
> ```cpp
>           model_instance_id_2);
>   const int num_vehicle_bodies_1 = vehicle_bodies_1.size();
>   const int num_vehicle_bodies_2 = vehicle_bodies_2.size();
> ```

Only one constant should be necessary; since the same SDF is used for both vehicles, the values should EXPECT_EQ the same.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 161 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf-UZvLX5OIWG1mUTj:-KVf-UZvLX5OIWG1mUTk:bh7nrco) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L161)):*
> ```cpp
> 
>   // The tests below will not work if the model has less than two bodies.
>   EXPECT_GE(num_vehicle_bodies_1, 2);
> ```

If "tests below will not work", then EXPECT should be ASSERT.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 176 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf05yriV7actOL_yJx:-KVf05ysvejlABBGEqck:b-q3ho2k) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L176)):*
> ```cpp
>   // Verifies that the first car was added to the tree.
>   EXPECT_EQ(vehicle_bodies_1.at(0)->get_name(), tree.get_body(1).get_name());
>   EXPECT_EQ(vehicle_bodies_1.at(1)->get_name(), tree.get_body(2).get_name());
> ```

Does this second test tell us anything more than that RBT is behaving correctly?  (I can buy that the prior `at(0)` test might say something about how AddTrajectoryCarFromSdf() has stacked up the models, but since `vehicle_bodies_1` came from querying the RBT, it's substructure as related to RBT.get_body() is a property of RBT.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Checkpoint of partial address of @maddog-tri's comments.

---

Review status: 3 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[drake/automotive/automotive_demo.cc, line 29 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeoFXo8ieqnVPsX6ne:-KVf9ZsLszjDlyy0PZ1k:b-qu6n94) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_demo.cc#L29)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Consider using the `kSdfFile` naming convention for this constant (not strictly required by style-guide since it's not static storage). The filename cannot be changed (e.g., supplied by command-line arg) without considering other hard-coded assumptions about the model's geometry (in particular, size and float-joint location --- e.g., CreateTrajectoryParams() expects a "car" to have a particular length/width).  Hmm... no matter the name of this variable, it should have a warning comment about changing the value.
</details>
Good point. Done.

---

*[drake/automotive/automotive_simulator.h, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3Eo2RaPbPttCHu3w:-KVf9uK78t8XmaBXoZwy:b-dhpbab) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L55)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> There are conditions on the content of the SDF file that should be documented here, e.g., single model, only the first/floating joint is controlled (others set to zeros), etc.
</details>
Done. The only assumption I can think of right now is that the SDF contains a single model. I looked at `drake::automotive::CreateTrajectoryParams()` and couldn't find documentation of its assumptions regarding vehicle dimensions. Do you know what assumptions are made by this method?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVfAYZZzIY_vYK1V3Kt:b-uhqqqg)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
Review status: 3 of 4 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[drake/automotive/automotive_simulator.h, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3Eo2RaPbPttCHu3w:-KVfYEGEq_Qd9jUF_okU:bxuo6ls) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L55)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Done. The only assumption I can think of right now is that the SDF contains a single model. I looked at `drake::automotive::CreateTrajectoryParams()` and couldn't find documentation of its assumptions regarding vehicle dimensions. Do you know what assumptions are made by this method?
</details>
("Done"?  reviewable is not showing me any change here yet.)  

I can't think of any more assumptions than what I wrote above:  single model with only a single operated floating joint.  `CreateTrajectoryParams()` has assumptions about the size/scale of the vehicle implicit in the radius and inset of the curves it creates --- e.g., the comment `// Magic car placement to make a good visual demo.` --- but that is not something that `AddSimpleCarFromSdf()` is concerned with (since `CreateTrajectoryParams()` is called by `automotive_demo.cc`, which is also choosing the SDF).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
@maddog-tri I addressed your comments. PTAL. thanks.

---

Review status: 1 of 6 files reviewed at latest revision, 11 unresolved discussions.

---

*[drake/automotive/automotive_simulator.h, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3Eo2RaPbPttCHu3w:-KVfgOnzYd1zJ1hwprat:b-v7qcxm) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L55)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> ("Done"?  reviewable is not showing me any change here yet.)  
> 
> I can't think of any more assumptions than what I wrote above:  single model with only a single operated floating joint.  `CreateTrajectoryParams()` has assumptions about the size/scale of the vehicle implicit in the radius and inset of the curves it creates --- e.g., the comment `// Magic car placement to make a good visual demo.` --- but that is not something that `AddSimpleCarFromSdf()` is concerned with (since `CreateTrajectoryParams()` is called by `automotive_demo.cc`, which is also choosing the SDF).
</details>
Sorry, I forgot to push my commits. It should be done now.

---

*[drake/automotive/automotive_simulator.h, line 67 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf3eQm_sCryobalS_6:-KVfiLqHwg29AMqq2EKp:b-civmuw) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L67)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Same comment as on sdf_filename param in AddSimpleCar().
</details>
Done. I modified this doc to point to `AddSimpleCarFromSdf()`'s doc.

---

*[drake/automotive/automotive_simulator.h, line 77 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeyzPsgOUflhG6iMOT:-KVfiiKickNgNIWIHUbh:b-pdjgo0) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/automotive_simulator.h#L77)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Does the return value serve any purpose beyond the unit-tests?
</details>
It is currently only used by the unit tests.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeww4yJLqiqyDacC-B:-KVfk10k2SV0cjhvUw9p:b-32n6zd) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L37)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> This parameterization is a lie:  the test code still depends on the Prius SDF ...or, at least an SDF that has a "chassis floor".  Also, the parameterization and restructuring as a helper function is a detriment unless the test is meaningfully invoked on more than the single Prius SDF.
</details>
I modified the unit test do not rely on "chassis floor" and added two additional tests using different SDFs.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 43 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVetuabyT2Nyht9sUQI:-KVfo3hzY5N345abK3sQ:boyexbg) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L43)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> This comment looks like it belongs with the definition of kJointStateChannelName.
</details>
Done. Thanks.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 57 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVeuDqnxlk1ELOrOJV1:-KVfjUOIWdq_ANf89JyV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L57)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Stale comment?  E.g., s/GetSystemByName/GetBuilderSystemByName/  ?
</details>
Done.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 70 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVexbA4mKtYyCznNRbX:-KVfq7OWOhhYdkMGJtyT:b-d7my8x) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L70)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> It's not completely clear to me what this is testing. Since `num_vehicle_bodies` came from a query to the tree, is it that we are making sure that AddSimpleCarFromSdf() didn't add something else?  It was more clear with `kNumVehicleBodies`; then the tester had an explicit expectation of what as in the SDF and what would end up in the RBT.
</details>
Yes, this test ensures that nothing else was surreptitiously added to the tree. I made the number of bodies a  method parameter that's hard coded by the unit test.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 134 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf4p7x8XOO1MzXy3ym:-KVfoDZvLQcZ4M-VKmSE:b-lmn5c8) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L134)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Same comment as TestSimpleCarWithSdf():  making it a helper function is just more noise until more than a single SDF is actually meaningfully tested.
</details>
Done. It's now tested with a two additional SDF models.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 158 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf-8Bt6tIdBUm775E2:-KVfoXiRY560dAOMnVdl:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L158)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Only one constant should be necessary; since the same SDF is used for both vehicles, the values should EXPECT_EQ the same.
</details>
Done.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 161 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf-UZvLX5OIWG1mUTj:-KVfohAKkuhHgGrfZlnr:boyexbg) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L161)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> If "tests below will not work", then EXPECT should be ASSERT.
</details>
Done. Thanks.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 176 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf05yriV7actOL_yJx:-KVfovnoJeMt-CklNXlR:b-pgdggo) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L176)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Does this second test tell us anything more than that RBT is behaving correctly?  (I can buy that the prior `at(0)` test might say something about how AddTrajectoryCarFromSdf() has stacked up the models, but since `vehicle_bodies_1` came from querying the RBT, it's substructure as related to RBT.get_body() is a property of RBT.)
</details>
You're right. The second test only tells us that RBT is behaving as expected. I removed the second check here and below.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVfsXugmUviERJ7AVhA:bf0pywg)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
Reviewed 1 of 1 files at r3, 4 of 5 files at r4.
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 134 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf4p7x8XOO1MzXy3ym:-KVg2z5XKAzL9ihUZ_s9:b36gywv) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L134)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Done. It's now tested with a two additional SDF models.
</details>
Last nit:  add a `num_vehicle_bodies` parameter, as with TestSimpleCarWithSdf(). 

In fact, add two of them, and add a second sdf_file parameter, and invoke the helper with a pair of different files, to check how they are stacked.  (Though, none of this test really tests very much until the "Confirm appropriate draw messages are coming out" TODO is resolved.  It's the product of 'model stack' with 'mux' that matters.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/models/two_dof_robot.sdf, line 1 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6dD041iZaQrRd0Vp:-KVg6dD041iZaQrRd0Vq:b-m5z478) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/models/two_dof_robot.sdf#L1)):*
> ```
> <sdf version='1.6'>
> ```

I see no reason to copy this file; automotive is allowed to depend on `systems`.  Or is it different somehow?

---

*[drake/automotive/test/automotive_simulator_test.cc, line 147 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6Wk-ejTwIbuWDg3J:-KVg6Wk0UgTsSHJ5xbxg:b555bwr) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/test/automotive_simulator_test.cc#L147)):*
> ```cpp
> GTEST_TEST(AutomotiveSimulatorTest, SimpleCarTestTwoDofBot) {
>   TestSimpleCarWithSdf(GetDrakePath() +
>                        "/automotive/models/two_dof_robot.sdf", 3);
> ```

I don't understand (nor will a reader understand) what adding a test case using this SDF accomplishes.  Will we ever run an automotive simulation using it, for example?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
I addressed all reviewer comments. PTAL. Thanks.

---

Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/models/two_dof_robot.sdf, line 1 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6dD041iZaQrRd0Vp:-KVgQTyTa9MNkzJihA_p:bk6xuh9) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/models/two_dof_robot.sdf#L1)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> I see no reason to copy this file; automotive is allowed to depend on `systems`.  Or is it different somehow?
</details>
Note that the version in `systems` is a URDF whereas this is an SDF.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 134 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVf4p7x8XOO1MzXy3ym:-KVgSMwBIhF7PDyzbDee:b-kuhqui) ([raw file](https://github.com/robotlocomotion/drake/blob/703f5472e251926ddfd16f01758ab702f4596f39/drake/automotive/test/automotive_simulator_test.cc#L134)):*
<details><summary><i>Previously, maddog-tri wrote…</i></summary>
> Last nit:  add a `num_vehicle_bodies` parameter, as with TestSimpleCarWithSdf(). 
> 
> In fact, add two of them, and add a second sdf_file parameter, and invoke the helper with a pair of different files, to check how they are stacked.  (Though, none of this test really tests very much until the "Confirm appropriate draw messages are coming out" TODO is resolved.  It's the product of 'model stack' with 'mux' that matters.)
</details>
Done. Agreed, I will resolve that `TODO` in a future PR.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 147 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6Wk-ejTwIbuWDg3J:-KVgQgxkaubwTtSaoQ1X:bkgfh0t) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/test/automotive_simulator_test.cc#L147)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> I don't understand (nor will a reader understand) what adding a test case using this SDF accomplishes.  Will we ever run an automotive simulation using it, for example?
</details>
Not sure. I added the test just to stress test the code using yet another SDF -- figured the more SDFs we test the code with the better. Would you like me to remove it? I added a comment explaining its purpose.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVgT_T-5274U4xjin96:bblnx7v)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 147 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6Wk-ejTwIbuWDg3J:-KVgUg63oLw2QyFGT8VF:bted8xw) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/test/automotive_simulator_test.cc#L147)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Not sure. I added the test just to stress test the code using yet another SDF -- figured the more SDFs we test the code with the better. Would you like me to remove it? I added a comment explaining its purpose.
</details>
I guess my premise is thus: we only care if the code works with models we will actually use.  If we've already tested all of them, I'm not sure what beyond that two_dof_robot adds?  And if not, why not test those other automotive models instead of a new irrelevant one?  Anyway, marking resolved.  As per top post, all of this body-index logic goes away in the future so might as well get this PR in as-is IMO.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/test/automotive_simulator_test.cc, line 147 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-KVg6Wk-ejTwIbuWDg3J:-KVgea1ynM7E2uqjZPu7:b-s9xbmq) ([raw file](https://github.com/robotlocomotion/drake/blob/e6645330c33c6e3bb70301df4cc30136696eddb4/drake/automotive/test/automotive_simulator_test.cc#L147)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> I guess my premise is thus: we only care if the code works with models we will actually use.  If we've already tested all of them, I'm not sure what beyond that two_dof_robot adds?  And if not, why not test those other automotive models instead of a new irrelevant one?  Anyway, marking resolved.  As per top post, all of this body-index logic goes away in the future so might as well get this PR in as-is IMO.
</details>
Good point. I removed `two_dof_robot.sdf` and modified the unit test to use `prius.sdf`, which is the only other vehicle SDF in `drake/automotive/models/` that was not used in these unit tests (all other models are either URDFs or an SDF that represents an environment).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 1 of 2 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3959#-:-KVh66RNohnMuycm1h5U:bnfp4nl)*
<!-- Sent from Reviewable.io -->
